mock/api.js

```diff
@@ -1,381 +1,297 @@
-              },
-            },
-          },
+export const mockSearch = {
+  hits: {
+    total: DEFAULT_SIZE,
+    hits: new Array(DEFAULT_SIZE).fill().map((value, index) => ({
+      _id: index + 1,
+      _source: {
+        run: {
+          config: casual.description,
+          name: casual.word,
+          script: casual.word,
+          user: casual.name,
         },
+        '@metadata.controller_dir': casual.word,
       },
-      id: 'test_run_name',
-      primaryMetrics: {},
-    },
+    })),
   },
-  iterationParams: {
-    instances: [1],
-    message_size_bytes: [1],
-    primary_metric: ['test_measurement_title'],
-    test_type: ['stream'],
-    closest_sample: [1],
-    uid: ['test_measurement_id'],
-    measurement_type: ['test_measurement_type'],
-    measurement_title: ['test_measurement_title'],
+};
+
+export const mockSessions = {
+  data: {
+    urls: new Array(DEFAULT_SIZE).fill().map((value, index) => ({
+      id: index + 1,
+      config: '{}',
+      description: casual.description,
+      createdAt: casual.date(),
+    })),
   },
 };
 
-export const expectedClusterData = {
+export const mockSession = {
   data: {
-    test_measurement_title: [
-      {
-        instances: 1,
-        max_stddevpct: 1,
-        message_size_bytes: 1,
-        primary_metric: 'test_measurement_title',
-        test_type: 'stream',
-        cluster: {
-          test_run_name: 0.1,
-          name_test_run_name: 'sample1',
-          percent_test_run_name: 1,
-          cluster: 1,
-        },
-        test_run_name: {
-          sample: {
-            closest_sample: 1,
-            mean: 0.1,
-            stddev: 0.1,
-            stddevpct: 1,
-            uid: 'test_measurement_id',
-            measurement_type: 'test_measurement_type',
-            measurement_idx: 0,
-            measurement_title: 'test_measurement_title',
-            '@idx': 0,
-            name: 'sample1',
-          },
-          run: {
-            id: 'test_run_id',
-            controller: 'test_controller',
-            name: 'test_run_name',
-            script: 'test_script',
-            config: 'test_config',
-          },
-          benchmark: {
-            instances: 1,
-            max_stddevpct: 1,
-            message_size_bytes: 1,
-            primary_metric: 'test_measurement_title',
-            test_type: 'stream',
-          },
-        },
-      },
-    ],
+    createSession: {
+      id: casual.uid,
+      config: '{}',
+      description: casual.description,
+    },
   },
-  keys: {},
-  params: {},
+};
+
+export default {
+  'GET /controllers/months': mockIndices,
+  'POST /controllers/list': mockControllers,
+  'POST /datasets/list': mockResults,
+  'POST /datasets/detail': mockDetail,
+  'POST /datasets/toc': mockTableContents,
+  'POST /datasets/samples': mockSamples,
+  'POST /mappings': mockMappings,
+  'POST /search': mockSearch,
+  'POST /sessions/list': mockSessions,
+  'POST /sessions/create': mockSession,
```

Comment:
How are these URIs being interpreted? The server `/datasets/list` and `datasets/detail` APIs aren't in the server main yet; and while `/datasets/toc` and `/datasets/samples` are reasonable suggestions those server APIs aren't implemented; nor are native search and "sessions" (if such things will even exist in the current terms, which seems unlikely).

I guess I need to search to see where this is referenced, though; perhaps it'll make sense.

Reply:
You'll notice in our `services/` definitions we specify `MOCK_UI` which is a runtime environment variable that allows us to determine if the dashboard is being run in a mocked environment. If that is the case, we tell the dashboard to hit these "fake" endpoints which ultimately return the mocked response. Over time, we can substitute these "fake" endpoints with real endpoints to `pbench_server` so we don't have to say `MOCK_UI ? /sessions/list : /graphl` when we build our endpoint url before submitting a request.